base/essentials.jl

```diff
@@ -662,7 +662,7 @@ call obsolete versions of a function `f`.
 (The drawback is that `invokelatest` is somewhat slower than calling
 `f` directly, and the type of the result cannot be inferred by the compiler.)
 """
-function invokelatest(f, args...; kwargs...)
+function invokelatest(@nospecialize(f), @nospecialize args...; kwargs...)
     # We use a closure (`inner`) to handle kwargs.
     inner() = f(args...; kwargs...)
     Core._apply_latest(inner)
```

Comment:
While we're in the process of optimizing this, might it make sense to check for empty kwargs here and if so use `Core._apply_latest(f, args)`?